### PR TITLE
Fix restoring error in deploy-agent sample 

### PR DESCRIPTION
**Pull request by mzat-msft**
# Description

Recently, the sample stopped working. The terminal output when the error
occurred only mentioned a Rollout worker not healthy without really
specifying the underlying cause.
We were not able to identify the issue, but noticed that using a different
approach for restoring the agent that does not use unpickling works
correctly.
For this reason, we revert back to the restoring approach that registers the
environment before loading the agent.

We also make dependency versions consistent between the `getting-started-on-aml` and `deploy-agent` samples.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Run getting started sample, download checkpoints and run deploy agent sample.
I tried to run the two samples end2end with the following package versions:

```
gymnasium==0.26.3  # Only version supported by rllib
numpy==1.24.3
ray==2.3, 2.4, 2.5
torch==2.0.1
```

I am planning to open another PR that updates the package versions for the samples to the latest published ray.

# Checklist:

- [x] I have squashed my previous commits into one commit and added a __meaningful__ commit message.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: examples/deploy-agent/README.md [hunk 1]
```diff
@@ -29,9 +29,13 @@ This sample shows how to deploy a trained agent using ``ray``'s module
     - Change the variable ``CHECKPOINT_FOLDER`` to the name of the folder containing your RLlib checkpoints.
     - Please note that ``CHECKPOINT_FOLDER`` must be the name of the folder inside ``checkpoints`` that contains the trained agent.
     - Modify the `observation_space` and `action_space` variables to be the same as in your simulation environment used for training the agent.
```
Comment (jillianmclements) [inline examples/deploy-agent/README.md:31]: We can remove everything related to modifying `observation_space` and `action_space` if we're registering the sim. This info will be read from the sim.

Comment (mzat-msft) [reply examples/deploy-agent/README.md:31]: I actually went the route of creating a dummy env with the given spaces. In this way the user does not have to copy-paste the sim.
Check the ``restore.py`` file in ``platotk``.

File: examples/deploy-agent/README.md [hunk 2]
```diff
@@ -29,9 +29,13 @@ This sample shows how to deploy a trained agent using ``ray``'s module
...
+      registering the simulation environment at training.
 
-3. Change package versions in `requirements.txt` to match your training environment.
-    - For example, if your training environment has ray==2.3.0 and this file has ray==2.4.0, you need to change it to ray==2.3.0.
+3. Change package versions in `requirements.txt` to match your training
+   environment. Pay particular attention to the ``ray``, ``tensorflow``,
+   ``torch``, ``numpy`` versions you have in the training environment and
+   specify the same versions in ``requirements.txt``.
 
 4. Test locally:
```
Comment (jillianmclements) [inline examples/deploy-agent/README.md:40]: On step 4 (test locally), I get the following relative import error when I run the serve command in src:
from platotk.restore import restore_agent
ModuleNotFoundError: No module named 'platotk'

Comment (jillianmclements) [reply examples/deploy-agent/README.md:40]: I modified `requirements.txt` as you suggested and I think installation was successful (see output below). However, I'm still getting the same ModuleNotFoundError from above. 

Collecting git+https://github.com/Azure/plato.git@mzat/fix-deploy-agent (from -r requirements.txt (line 1))
  Cloning https://github.com/Azure/plato.git (to revision mzat/fix-deploy-agent) to /private/var/folders/90/6x51pxbn6vl_spycvr6r218w0000gn/T/pip-req-build-lp5wkqq4
  Running command git clone --filter=blob:none --quiet https://github.com/Azure/plato.git /private/var/folders/90/6x51pxbn6vl_spycvr6r218w0000gn/T/pip-req-build-lp5wkqq4
  Running command git checkout -b mzat/fix-deploy-agent --track origin/mzat/fix-deploy-agent
  Switched to a new branch 'mzat/fix-deploy-agent'
  branch 'mzat/fix-deploy-agent' set up to track 'origin/mzat/fix-deploy-agent'.
  Resolved https://github.com/Azure/plato.git to commit 071fd69404700ff4f510ca158a7a2083f82fa21d
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done

Comment (mzat-msft) [reply examples/deploy-agent/README.md:40]: I was able to reproduce.
Apparently you have to first remove the package and then reinstall it.
```bash
pip uninstall platotk
```

Comment (jillianmclements) [reply examples/deploy-agent/README.md:40]: Thanks, it works now. 

Side note: installing `tensorflow==2.11.1` on a mac M1 throws an error, so I'll open another PR with M1 instructions in the readme. Alternatively, we could switch to Ray 2.4+ which uses pytorch as the default instead.

**Review by jillianmclements — APPROVED** [review]
Great work! This runs on my M1.

**Pull request merged**